src/js/utils/MarathonUtil.js

```diff
@@ -1,12 +1,63 @@
+import VolumeTypes from '../constants/VolumeTypes'
+import VolumeStatus from '../constants/VolumeStatus'
+
 function parseApp(app) {
-  let {id} = app;
+  let {id, tasks = [], container = {}} = app;
 
-  if (!id.startsWith('/') || id.endsWith('/')) {
+  if (id == null || !id.startsWith('/') || id.endsWith('/')) {
     throw new Error(`Id (${id}) must start with a leading slash ("/") ` +
       'and should not end with a slash.');
   }
 
-  return app;
+  if (container == null || container.volumes == null || !Array.isArray(container.volumes)) {
```

Comment:
This first check is unnecessary, since you are default assigning it to an object, unless it can be `null`, but why would it?